src/watertap_contrib/seto/multiperiod/VAGMD_batch_flowsheet.py

```diff
@@ -0,0 +1,455 @@
+        doc="Specific thermal power consumption (kWh/m3)",
+    )
+
+    m.fs.specific_energy_consumption_electric = Var(
+        initialize=1,
+        units=pyunits.kWh / pyunits.m**3,
+        doc="Specific electric power consumption (kWh/m3)",
+    )
+
+    m.fs.gain_output_ratio = Var(
+        initialize=10,
+        units=pyunits.dimensionless,
+        doc="Gain output ratio",
+    )
+
+    """
+    Constriant equatiosn
+    """
+    mfs = m.fs
+
+    @mfs.Constraint(doc="Calculate time interval of each period")
+    def eq_dt(b):
+        if module_type == "AS7C1.5L":
+            return b.dt == 20352.55 / feed_flow_rate
+        else:  # module_type == "AS26C7.2L"
+            return b.dt == 73269.19 / feed_flow_rate
+
+    @mfs.Constraint(doc="Calculate current feed salinity")
+    def eq_feed_salinity(b):
+        return b.vagmd.feed_props[0].conc_mass_phase_comp["Liq", "TDS"] * (
+            initial_batch_volume - b.acc_distillate_volume
+        ) == b.pre_feed_salinity * (initial_batch_volume - b.pre_acc_distillate_volume)
+
+    @mfs.Constraint(doc="Calculate current feed temperature")
+    def eq_feed_temp(b):
+        return b.vagmd.feed_props[0].temperature == (
+            feed_flow_rate
+            * pyunits.convert(b.dt, to_units=pyunits.h)
+            * b.pre_evap_out_temp
+            + (initial_batch_volume - b.pre_acc_distillate_volume)
+            * b.pre_feed_temperature
+        ) / (
+            feed_flow_rate * pyunits.convert(b.dt, to_units=pyunits.h)
+            + initial_batch_volume
+            - b.pre_acc_distillate_volume
+        )
+
+    @mfs.Constraint(doc="Calculate accumulated distillate volume")
+    def eq_acc_distillate_volume(b):
+        return b.acc_distillate_volume == b.pre_acc_distillate_volume + pyunits.convert(
+            b.pre_permeate_flow_rate * b.dt, to_units=pyunits.L
+        )
+
+    @mfs.Constraint(doc="Calculate accmulated recovery ratio")
+    def eq_acc_recovery_ratio(b):
+        return (
+            b.acc_recovery_ratio
+            == 1
+            - feed_salinity / b.vagmd.feed_props[0].conc_mass_phase_comp["Liq", "TDS"]
+        )
+
+    @mfs.Constraint(doc="Calculate accmulated thermal energy consumption")
+    def eq_acc_thermal_energy(b):
+        return b.acc_thermal_energy == b.pre_acc_thermal_energy + pyunits.convert(
+            b.pre_thermal_power * b.dt, to_units=pyunits.kWh
+        )
+
+    @mfs.Constraint(doc="Calculate accmulated thermal energy consumption")
+    def eq_acc_cooling_energy(b):
+        return b.acc_cooling_energy == b.pre_acc_cooling_energy + pyunits.convert(
+            b.pre_cooling_power * b.dt, to_units=pyunits.kWh
+        )
+
+    @mfs.Constraint(doc="Calculate specific thermal energy consumption")
+    def eq_specific_energy_consumption_thermal(b):
+        return b.specific_energy_consumption_thermal == pyunits.convert(
+            b.acc_thermal_energy / (b.acc_distillate_volume + 1e-8 * pyunits.L),
```

Comment:
My guess is to avoid division by 0. If that's the case, we could try just multiplying the left-hand side by `acc_distillate_volume` instead of letting it sit in the denominator.

Reply:
Yes, it is to avoid 0 in the denominator. I tried a few times to move it to the left-hand side as well, but then the model wouldn't converge no matter what scaling factor I assigned to this constraint.  
Do you think we can keep it this way for now and investigate it in future PRs?